fe/fe-core/src/main/java/com/starrocks/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -471,8 +471,8 @@ private void predicateBaseAndCheck(Predicate node) {
             node.setType(Type.BOOLEAN);
 
             for (Expr expr : node.getChildren()) {
-                if (expr.getType().isOnlyMetricType()) {
-                    throw new SemanticException("HLL, BITMAP and PERCENTILE type couldn't as Predicate");
+                if (expr.getType().isOnlyMetricType() || expr.getType() instanceof ArrayType) {
```

Comment:
How about move this logical to `Type.java` ?

Reply:
ok